### ISSUE TITLE
Run in remote ignores custom port

### ISSUE THREAD

**Issue by albertotb** [guildai/guildai#139]
When trying to run a script in a remote, I get the following error (hostname changed for security reasons):

```
ssh: connect to host my.hostname port 22: Connection refused
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(226) [sender=3.1.1]
Traceback (most recent call last):
  File "/home/alberto/miniconda3/bin/guild", line 8, in <module>
    sys.exit(main())
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/main_bootstrap.py", line 38, in main
    guild.main.main()
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/main.py", line 32, in main
    _main()
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/main.py", line 38, in _main
    main_cmd.main(standalone_mode=False)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/external/click/core.py", line 759, in __call__
    return self.main(*args, **kwargs)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/external/click/core.py", line 714, in main
    rv = self.invoke(ctx)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/external/click/core.py", line 1132, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/external/click/core.py", line 951, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/external/click/core.py", line 552, in invoke
    return callback(*args, **kwargs)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/click_util.py", line 146, in fn
    return fn0(*(args + (Args(**kw),)))
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run.py", line 351, in run
    run_impl.main(args)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run_impl.py", line 68, in main
    _dispatch_cmd(args, opdef)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run_impl.py", line 419, in _dispatch_cmd
    _dispatch_op_cmd(opdef, args)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run_impl.py", line 522, in _dispatch_op_cmd
    _maybe_run(op, args)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run_impl.py", line 996, in _maybe_run
    _run(op, args)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run_impl.py", line 1107, in _run
    _run_remote(op, args)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/run_impl.py", line 1117, in _run_remote
    remote_impl_support.run(args)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/commands/remote_impl_support.py", line 98, in run
    run_id = remote.run_op(**_run_kw(args))
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/remotes/ssh.py", line 156, in run_op
    remote_run_dir = self._init_remote_run(tmp.path, opspec, restart)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/remotes/ssh.py", line 170, in _init_remote_run
    self._copy_package_dist(package_dist_dir, remote_run_dir)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/remotes/ssh.py", line 234, in _copy_package_dist
    port=self.port)
  File "/home/alberto/miniconda3/lib/python3.7/site-packages/guild/remotes/ssh_util.py", line 90, in rsync_copy_to
    subprocess.check_call(cmd)
  File "/home/alberto/miniconda3/lib/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['rsync', '-vr', '-e', 'ssh -oConnectTimeout=5555', '/tmp/guild-remote-pkg-67wkome2/', 'my_user@my.hostname:.guild/runs/b470b6a154624b07ad74c09aaf99ae96/.guild/job-packages/']' returned non-zero exit status 255.
```

In the configuration file I have:

```
remotes:
    test:
        type: ssh
        host: my.hostname
        port: 5555
        user: my_user
        public-key: ~/.ssh/id_rsa.pub
```

but it seems to be ignoring the `port` option:

```
ssh: connect to host my.hostname port 22: Connection refused
```

and instead passing it as the `ConnectTimeout` option

```
subprocess.CalledProcessError: Command '['rsync', '-vr', '-e', 'ssh -oConnectTimeout=5555', '/tmp/guild-remote-pkg-67wkome2/', 'my_user@my.hostname:.guild/runs/b470b6a154624b07ad74c09aaf99ae96/.guild/job-packages/']
```

Edit: I found issue #28, which seems related, so I checked my guild version:

```
guild_version:             0.6.6
guild_install_location:    /home/alberto/miniconda3/lib/python3.7/site-packages/guild
guild_home:                /home/alberto/miniconda3/.guild
guild_resource_cache:      /home/alberto/miniconda3/.guild/cache/resources
installed_plugins:         cpu, disk, exec_script, gpu, keras, memory, perf, python_script, skopt
python_version:            3.7.4 (default, Aug 13 2019, 20:35:49) [GCC 7.3.0]
python_exe:                /home/alberto/miniconda3/bin/python
platform:                  Linux 4.4.0-18362-Microsoft x86_64
tensorboard_version:       2.1.1
cuda_version:              not installed
nvidia_smi_version:        not installed
latest_guild_version:      0.6.6
```

**Comment by gar1t** [comment]
I believe this is resolved in the pre-release version. Could you upgrade and try that?

    $ pip install guildai --upgrade --pre

Note the `--pre` option - that'll make sure you get the latest!

**Comment by albertotb** [comment]
Yes, the pre-release version solved the `ssh` issue but now it cannot find the model. I will open another one.

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0. If this is still an issue, please re-open and we'll sort it out. Thanks again for the report!
